src/flyte/_internal/runtime/task_serde.py

```diff
@@ -239,14 +239,14 @@ def _get_urun_container(
     resources = get_proto_resources(task_template.resources)
 
     img = task_template.image
-    if isinstance(img, str):
-        raise flyte.errors.RuntimeSystemError("BadConfig", "Image is not a valid image")
-
     env_name = task_template.parent_env_name
     if env_name is None:
         raise flyte.errors.RuntimeSystemError("BadConfig", f"Task {task_template.name} has no parent environment name")
 
-    img_uri = lookup_image_in_cache(serialize_context, env_name, img)
+    if isinstance(img, flyte.Image):
+        img_uri = lookup_image_in_cache(serialize_context, env_name, img)
+    else:
```

Comment:
lets not deal with strings at all in this layer. lets look for image has no layers then just use the base image
if this is not true we should raise an error